src/widget.js

```diff
@@ -1,23 +1,140 @@
+              <img src="../img/v1-icon.png" />
+              <span>Donate Now</span>
+            </a>
+          </div>`
+};
+
+class SuperheroButton {
+  
+  constructor(element, options) {
+    const opt = { ...defaults, ...options };
+    this.insertStyles(opt);
+    const el = this.isDomNode(element) || this.isNodeList(element) ? element : opt.all ? document.querySelectorAll(element) : document.querySelector(element);
+    if (this.isNodeList(el)) {
+      let buttonInstances = [];
+      el.forEach(e => {
+        const instance = this.createButtonInstance(opt);
+        buttonInstances.push(instance);
+        e.replaceWith(instance);
+      });
+      this.instance = buttonInstances;
+    } else {
+      this.instance = this.createButtonInstance(opt);
+      el.replaceWith(this.instance);
+    }
+  }
+
+  isDomNode(element) {
+    return element instanceof Element || element instanceof HTMLDocument
   }
 
-  for(var f in iframes) {
-    document.querySelectorAll("a.superhero-tip-button")[0].replaceWith(iframes[f]);
+  isNodeList(nodes) {
+    return NodeList.prototype.isPrototypeOf(nodes);
   }
-});
\ No newline at end of file
+
+  getInstance() {
+    return this.instance;
+  }
+
+  createButtonInstance({ size, url, account }) {
+    if (!templates[size]) throw new Error('Unsupported size');
+    var button = document.createElement('button');
+    button.innerHTML = templates[size];
+    button.className = `superhero-button-container superhero-button-${size}`;
+    const isMobile = navigator.userAgent.match(/(iPad)|(iPhone)|(iPod)|(android)|(webOS)/i);
+    let deeplink = new URL('https://wallet.superhero.com/tip');
+    if (isMobile) {
+      
+      deeplink.searchParams.set('url', encodeURIComponent(url));
+      deeplink.searchParams.set('x-success', url);
+      deeplink.searchParams.set('x-cancel', url);
+    } else {
+      const storeLinks = {
+        firefox: 'https://addons.mozilla.org/en-US/firefox/addon/superhero-wallet/',
+        chrome: 'https://chrome.google.com/webstore/detail/superhero/mnhmmkepfddpifjkamaligfeemcbhdne'
+      };
+      const { name } = detect();
+      if (storeLinks[name]) deeplink = new URL(storeLinks[name]);
+    }
+    
+    var link = button.querySelector("#superhero-tip-link");
+    link.href = deeplink;
+
+    // add data-account atribute for claiming
+    if (account) link.setAttribute("data-account", account);
+    // add data-url atribute to be detected by wallet extension
+    link.setAttribute("data-url", url);
+
+    fetch('https://raendom-backend.z52da5wt.xyz/cache/stats')
+    .then(res => res.json())
+    .then(res => {
+      const tips = res.by_url.find(u => u.url === url);
+      var tipsEl = button.querySelector("#tips");
+      if (tipsEl) tipsEl.innerHTML = tips ? tips.total_amount : 0;
+    });
+
+    return button;
+  }
+
+  insertStyles({ inlineCss }) {
+    let css;
+    // TODO - implment the logic for inserting the button styles
+    if (inlineCss) {
```

Comment:
```suggestion
    if (process.env.INLINE_CSS) {
```
This switch I proposing to implement using webpack define plugin and to have two builds: in the first one to inline css into the head (by default) and in other to do nothing with css (to allow a developer to manage styles by himself).